backend/src/routes/v2/secrets.ts

```diff
@@ -0,0 +1,177 @@
+                } 
+            } else {
+                throw new Error('secrets must be an object or an array of objects')
+            }
+            
+            return true;
+    }),
+    validateRequest,
+    requireAuth({
+        acceptedAuthModes: ['jwt']
+    }),
+    requireWorkspaceAuth({
+        acceptedRoles: [ADMIN, MEMBER],
+        location: 'body'
+    }),
+    secretsController.createSecrets
+);
+
+router.get(
+    '/',
+    query('workspaceId').exists().trim(),
+    query('environment').exists().trim().isIn(['dev', 'staging', 'prod', 'test']),
+    validateRequest,
+    requireAuth({
+        acceptedAuthModes: ['jwt', 'serviceToken']
+    }),
+    requireWorkspaceAuth({
+        acceptedRoles: [ADMIN, MEMBER],
+        location: 'query'
+    }),
+    secretsController.getSecrets
+);
+
+router.patch(
+    '/',
+    body('workspaceId').exists().trim(),
+    body('environment').exists().trim().isIn(['dev', 'staging', 'prod', 'test']),
+    body('secrets')
+        .exists()
+        .custom((value) => {
+            if (Array.isArray(value)) {
+                // case: update multiple secrets
+                if (value.length === 0) throw new Error('secrets cannot be an empty array')
+                for (const secret of value) {
+                    if (
+                        !secret.id || 
+                        !secret.secretKeyCiphertext ||
+                        !secret.secretKeyIV ||
+                        !secret.secretKeyTag ||
+                        !secret.secretValueCiphertext ||
+                        !secret.secretValueIV ||
+                        !secret.secretValueTag
+                    ) {
+                        throw new Error('secrets array must contain objects that conform to the Secret interface');
+                    }
+                }
+            } else if (typeof value === 'object') {
+                // case: update 1 secret
+                if (
+                    !value.id || 
+                    !value.secretKeyCiphertext ||
+                    !value.secretKeyIV ||
+                    !value.secretKeyTag ||
+                    !value.secretValueCiphertext ||
+                    !value.secretValueIV ||
+                    !value.secretValueTag
+                ) {
+                    throw new Error('secrets array must contain objects that conform to the Secret interface');
+                } 
+            } else {
+                throw new Error('secrets must be an object or an array of objects')
+            }
+            
+            return true;
+    }),
+    validateRequest,
+    requireAuth({
+        acceptedAuthModes: ['jwt']
+    }),
+    requireWorkspaceAuth({
+        acceptedRoles: [ADMIN, MEMBER],
+        location: 'body'
+    }),
+    requireSecretsAuth({
+        acceptedRoles: [ADMIN, MEMBER]
+    }),
+    secretsController.updateSecrets
+);
+
+router.delete(
+    '/',
+    [
```

Comment:
Is this correct? The check is inside of Array but the Array is not deconstructed. Should the correct form be `...[check('secretIds')]`

Reply:
Yeah I believe it's correct. I've tested it with and without correct `secretIds`

Reply:
Oh actually hmm maybe it would also work without the array..

Reply:
Alrighty fixed.